### PR TITLE
Address Changes Requested for v3.0.0

### PR DESCRIPTION
**Pull request by ifl0w**
This PR mainly collects changes required to pass the [review on extensions.gnome.org](https://extensions.gnome.org/review/51024).

### CODE REVIEW

**Comment by ifl0w** [comment]
Hey @Lucki, may I ask you to take a quick look at the [second commit](https://github.com/ifl0w/RandomWallpaperGnome3/pull/189/commits/3c84cd2f1bdba56231558e8a6e3b683f43e0726e) and maybe test this PR to see if you recognize any major issues?

I tried to test some edge cases manually, and for me, it seems to work, but I might not have tested cases that require the earlier logic.

**Review by Lucki — CHANGES_REQUESTED** [review]
~~I still don't know why my hover preview doesn't work, but from the debug message I can see that it wants to do the correct stuff.~~

I think I just found out why my hover preview doesn't work anymore: I'm using dark mode and we must have lost setting the `picture-uri-dark` somewhere.

File: src/randomWallpaperMenu.ts [hunk 1]
```diff
@@ -261,25 +239,10 @@ class RandomWallpaperMenu {
...
-                if (this._settings.getInt('change-type') as Mode === Mode.LOCKSCREEN && this._savedBackgroundUri) {
-                    // Reset background after previewing the lock screen options
-                    this._wallpaperController.resetWallpaper(this._savedBackgroundUri);
-                } else {
-                    // Update saved background with newly set background image
-                    // so we don't revert to an older state when closing the menu
-                    const backgroundSettings = new Settings.Settings('org.gnome.desktop.background');
-                    this._savedBackgroundUri = backgroundSettings.getString('picture-uri');
-                }
-
```
[the remark this answers is not in the document]
Comment (ifl0w) [reply src/randomWallpaperMenu.ts]: Good catch, thanks! 

I simply reset the preview before setting the wallpaper in the fixup commit now. That seems to work for me when artificially prolonging the defaultWallpaperManager. Can you verify?

Note: This isn't really a nice solution and maybe I should take another look at reworking the logic to prevent this kind of race conditions altogether. But for now, I'd be fine with this quick fix.

**Comment by ifl0w** [comment]
> I think I just found out why my hover preview doesn't work anymore: I'm using dark mode and we must have lost setting the picture-uri-dark somewhere.

@Lucki, I understand that this could be a problem in general as there are a few places where we use `picture-uri`, and it looks like we might not consider the `picture-uri-dark` correctly. However, the hover preview still works for me in the dark UI mode. Are you sure it is only due to the dark mode in your case?

**Comment by Lucki** [comment]
> However, the hover preview still works for me in the dark UI mode. Are you sure it is only due to the dark mode in your case?

Sorry, I only saw one entry change in dconf, might have been because the HydraPaper process got killed. I should have tested light mode, it also doesn't work for me there.

Debugging is particularly annoying since Gnome 45 because the gnome shell crashes instead of reloading. I'll give that workaround a try later.

File: src/prefs.ts [hunk 2]
```diff
@@ -43,6 +46,11 @@ class RandomWallpaperSettings extends ExtensionPreferences {
         window.set_search_enabled(true);
 
         backendConnection.setBoolean('pause-timer', true);
+        // save and observe the pause-timer state while settings are open
+        this.pauseTimerState = backendConnection.getBoolean('pause-timer');
+        backendConnection.observe('pause-timer', () => {
+            this.pauseTimerState = backendConnection.getBoolean('pause-timer');
+        });
```
[the remark this answers is not in the document]
Comment (ifl0w) [reply src/prefs.ts:53]: > With this change it's always pausing the timer because of the ordering.

Uff :facepalm: thanks!

> My original intention was that the timer doesn't fire while changes are being made.

That makes kind of sense to me, but I am not sure how important or confusing this functionality is in the end. I'd just leave it in for now. We can remove it at any time if we decide to.

> If we're going further with this idea, we ideally would disable the pause switch while the pref menu is open to prevent switching the timer on with the menu open. This also would remove the need to observe the current state.

I'm not really convinced that this is better from a user perspective without some clear and visible documentation on this behavior. I think I'd rather remove this functionality entirely if we run into any problems down the road.

**Comment by ifl0w** [comment]
> Debugging is particularly annoying since Gnome 45 because the gnome shell crashes instead of reloading. I'll give that workaround a try later.

I feel your pain! For me, gnome shell on x-org crashes when restarting it :sweat_smile: ... the linux container is currently my only debug/test workflow on my main machine.

File: src/randomWallpaperMenu.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lucki) [reply src/randomWallpaperMenu.ts]: Yes, works.

**Comment by Lucki** [comment]
Many `console.log()` later: #190

**Review by Lucki — APPROVED** [review]
